synapse/storage/databases/main/event_federation.py

```diff
@@ -1599,18 +1651,20 @@ async def get_all_rooms_with_staged_incoming_events(self) -> List[str]:
         )
 
     @wrap_as_background_process("_get_stats_for_federation_staging")
-    async def _get_stats_for_federation_staging(self):
+    async def _get_stats_for_federation_staging(self) -> None:
         """Update the prometheus metrics for the inbound federation staging area."""
 
-        def _get_stats_for_federation_staging_txn(txn):
+        def _get_stats_for_federation_staging_txn(
+            txn: LoggingTransaction,
+        ) -> Tuple[int, int]:
             txn.execute("SELECT count(*) FROM federation_inbound_events_staging")
-            (count,) = txn.fetchone()
+            (count,) = cast(Tuple[int], txn.fetchone())
 
             txn.execute(
                 "SELECT min(received_ts) FROM federation_inbound_events_staging"
             )
 
-            (received_ts,) = txn.fetchone()
+            (received_ts,) = cast(Tuple[int], txn.fetchone())
```

Comment:
I don't think this cast is correct. If there are now rows selected, I think this query returns NULL(?)

Reply:
I have changed to `Tuple[Optional[int]]`